agent/src/protocols/http/discovery_handler.rs

```diff
@@ -0,0 +1,79 @@
+// TODO(dazwilkin) Why is aliasing required for the lambda?
+use super::super::{DiscoveryHandler as DH, DiscoveryResult as DR};
+
+use akri_shared::akri::configuration::HTTPDiscoveryHandlerConfig;
+use async_trait::async_trait;
+use failure::Error;
+use reqwest::get;
+use std::collections::HashMap;
+
+const BROKER_NAME: &str = "AKRI_HTTP";
+const DEVICE_ENDPOINT: &str = "AKRI_HTTP_DEVICE_ENDPOINT";
+
+pub struct HTTPDiscoveryHandler {
+    discovery_handler_config: HTTPDiscoveryHandlerConfig,
+}
+impl HTTPDiscoveryHandler {
+    pub fn new(discovery_handler_config: &HTTPDiscoveryHandlerConfig) -> Self {
+        println!("[http:new] Entered");
+        HTTPDiscoveryHandler {
+            discovery_handler_config: discovery_handler_config.clone(),
+        }
+    }
+}
+#[async_trait]
+
+impl DH for HTTPDiscoveryHandler {
+    async fn discover(&self) -> Result<Vec<DR>, failure::Error> {
+        println!("[http:discover] Entered");
+
+        let url = self.discovery_handler_config.discovery_endpoint.clone();
+        println!("[http:discover] url: {}", &url);
```

Comment:
I would be consistent and use tracing instead of prints.

Reply:
Fixed: https://github.com/deislabs/akri/pull/135/commits/3d14e3acec60d38a47d7885e717f2722faf4bf79